### PR TITLE
[TC2] Added comment quality check for notifications

### PR DESCRIPTION
**Pull request by marksivan**
### Why the change
After a discussion with manager, we decided to increase the complexity and relevance of reply-to-comment notifications. Notifications will now only be sent if replies meet content quality standards — specifically, they must contain at least 5 words (excluding emojis) and must not contain profanity.

### Key Changes:
- Created commentFilters.js utility module with three core functions:

  - containsBadWords(comment): Uses the bad-words package to check for profanity.

  - countWords(comment): Counts words while ignoring emojis.

 - shouldCreateNotification(comment): Combines the above checks to determine if a reply qualifies for notification.

- Updated comments.js route to integrate these filters before triggering a notification for replies.

- Added test script to validate the filtering logic with a variety of edge cases (e.g., emoji-only, short comments, profanity).

- Implemented fallback mechanism: Ensures that even if bad-words fails or becomes incompatible, the system won’t break   it logs the issue and bypasses the profanity check gracefully.

- Created see-bad-words.js: A utility to log and review the word list from the bad-words package for internal moderation reference.

### CODE REVIEW

File: backend/utils/commentFilters.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+const containsBadWords = (text) => {
+  if (!text || typeof text !== 'string') return false;
+
+  try {
+    if (filter) {
+      return filter.isProfane(text);
+    } else {
+      // Fallback if filter initialization failed
+      const commonBadWords = ['fuck', 'shit', 'ass', 'bitch', 'damn', 'foolish'];
+      return commonBadWords.some(word => text.toLowerCase().includes(word));
```
Comment (cissygao) [inline backend/utils/commentFilters.js:31]: Please consider using a trie to increase efficiency.

Comment (marksivan) [reply backend/utils/commentFilters.js:31]: Noted. Thank you

Comment (marksivan) [reply backend/utils/commentFilters.js:31]: Resolved [here](https://github.com/Story-Stack/story-stack-capstone/pull/50)

File: backend/test-comment-filter.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      expected: {
+        hasBadWords: true,
+        wordCount: 6,
+        shouldNotify: false, // Has bad words (should not notify regardless of length)
+      },
+    },
+  ];
+
+  // Run tests
+  console.log("TESTING COMMENT FILTER FUNCTIONALITY\n");
```
Comment (daphneyang3) [inline backend/test-comment-filter.js:79]: nit: remove testing logs

Comment (marksivan) [reply backend/test-comment-filter.js:79]: Sure thing
